### PR TITLE
Add critical USER_UNAUTHORIZED_ENDPOINT and enable clinicians to generate adherence reports

### PR DESCRIPTION
**Pull request by pbugni**
Clinicians were blocked from accessing `/api/report/questionnaire_status` (an easy fix) but this was difficult to track as we didn't have a `USER_UNAUTHORIZED_ENDPOINT` configured, and would therefore simply redirect to '' and return a 301 status rather than a 401 as needed for the front end to be aware of the error situation.

### CODE REVIEW

**Review by achen2401 — APPROVED** [review]
Thanks, Paul, for the fix.  The changes look good to me - I didn't encounter error when exporting report data as a user with just the clinician role.  Also, I tested the error scenario (e.g., unauthorized use), that looks good as well as error is displayed on the frontend (see screenshot): 

Screen Shot 2021-02-11 at 2 39 34 PM

File: portal/views/portal.py [hunk 1]
```diff
@@ -141,6 +141,8 @@ def report_slow_queries(response):
 
     """
     if current_app.config.get("LOG_SLOW_RESPONSES"):
+        if not hasattr(g, 'start_request_time'):
+            return response
```
Comment (ivan-c) [inline portal/views/portal.py:145]: This doesn't look related to the bugfix. If I understand correctly, this seems to exit early if the request is not being timed, but isn't that controlled by the above config?

File: portal/views/auth.py [hunk 2]
```diff
@@ -1025,3 +1025,21 @@ def authorize(*args, **kwargs):
         return redirect('/')
     # See "hardwired" note in docstring above
     return True
+
+
+@auth.route('/unauthorized_endpoint')
+@oauth.require_oauth()
```
Comment (ivan-c) [inline portal/views/auth.py:1031]: Do we need this decorator for an endpoint that is effectively informational? Worried it'll cause a redirect loop if flask-user is sending unauthenticated users here

**Review by ivan-c — APPROVED** [review]
Looks good! Had some questions, but ready to merge

File: portal/views/portal.py (hunk 1 above)
Comment (pbugni) [reply portal/views/portal.py:145]: it's just something that blew up on me when testing - never seen it before, but seemed worthy of adding a safeguard.

File: portal/views/auth.py (hunk 2 above)
Comment (pbugni) [reply portal/views/auth.py:1031]: that's a different config, USER_UNAUTHENTICATED_ENDPOINT.  i walked through the flask-user code and confirmed we only hit this when the user is authenticated (logged-in) but not authorized.
